framework/src/tinyalsa/tinyalsa.c

```diff
@@ -541,18 +541,12 @@ int pcm_readi(struct pcm *pcm, void *data, unsigned int frame_count)
 	bufdesc.session = pcm->session;
 #endif
 
-	/* If device is not yet started, start now! */
-	if (!pcm->running) {
-		pcm->enqueuedBytes = 0;
+	/* If no pending buffers ,then enqueue some buffers*/
+	if (pcm->enqueuedBytes==0) {
 		/* If the device is opened for read and Our buffers are not enqued. Enque them now. */
-#ifdef CONFIG_AUDIO_MULTI_SESSION
-		bufdesc.session = pcm->session;
-#endif
 #ifdef CONFIG_AUDIO_DRIVER_SPECIFIC_BUFFERS
-		bufdesc.numbytes = pcm->config.period_size;
 		for (pcm->bufPtr = 0; pcm->bufPtr < buf_info.nbuffers; pcm->bufPtr++)
 #else
```

Comment:
I mean you can use same code because pcm->config.period_count has number of buffer all the time.